solidity/contracts/XERC20Lockbox.sol

```diff
@@ -26,6 +27,22 @@ contract XERC20Lockbox is IXERC20Lockbox {
    */
   bool public immutable IS_NATIVE;
 
+  /**
+   * @notice The ERC20 decimals, assumed to be 0 if the token does not have a decimals() method
+   */
+  uint8 public erc20Decimals;
+
+  /// @dev If nativeTokenDecimals is different than 18 decimals, bridge will inflate or deflate token amounts
+  ///      when depositing to child chain to match 18 decimal denomination. Opposite process happens when
+  ///      amount is withdrawn back to parent chain. In order to avoid uint256 overflows we restrict max number
+  ///      of decimals to 36 which should be enough for most practical use-cases.
+  uint8 constant MAX_DECIMALS = uint8(36);
+
+  /// @dev Max amount of erc20 native token that can deposit when upscaling is required (i.e. < 18 decimals)
+  ///      Amounts higher than this would risk uint256 overflows when adjusting decimals. Considering
+  ///      18 decimals are 60 bits, we choose 2^192 as the limit which equals to ~6.3*10^57 weis of token
```

Comment:
With `MAX_DECIMALS` the factor can be as high as 10**18 (36 decimals - 18 decimals or (18 - 0).

This comes from `Math.log2(10**18) ~ 60 bits`, and 192 as 256-60, and a little less for safety reasons and practicality of `type(uint192)`.